ganalytics.cpp

```diff
@@ -473,10 +501,12 @@ GAnalytics::GAnalytics(const QString &trackingID, QObject *parent)
 }
 
 /**
- * Destructor of class GAnalytics.
+ * Destructor of class GAnalytics. Ends the session.
  */
 GAnalytics::~GAnalytics()
 {
+    if (!trackingID().isEmpty())
+        endSession();
```

Comment:
Waiting in the destructor is not a great idea since there is not really an option to wait except hacking around with QEventLoop.

There is no way to send the remaining events using the current API. You have to make ```GAnalytics::Private::postMessage()``` publicly accessible by forwarding it from the ```GAnalytics``` class. Then you can wait for the "isSendingChanged" signal to really quit like the demo application does: https://github.com/HSAnet/qt-google-analytics/blob/master/examples/console-app/main.cpp#L11